account_cost_spread/models/account_invoice_spread_line.py

```diff
@@ -55,6 +57,60 @@ class AccountInvoiceSpreadLine(models.Model):
         string='Initial Balance Entry',
         help="Set this flag for entries of previous fiscal years "
              "for which OpenERP has not generated accounting entries.")
+    can_create_move = fields.Boolean(compute='_compute_can_create_move')
+    sequence = fields.Integer(required=True, default=1)
+
+    @api.multi
+    def _check_invoice_number(self):
+        """ Check invoice number """
+        if not bool(filter(None, self.mapped(
+            'invoice_line_id.invoice_id.internal_number'
+        ))):
+            raise ValidationError(
+                _('Linked invoice has no internal number.'))
+
+    @api.multi
+    def _check_invoice_state(self):
+        """ Check invoice state """
+        self.ensure_one()
+        state = self.mapped('invoice_line_id.invoice_id.state')[0]
+        if state not in ('open', 'paid'):
+            raise ValidationError(_('Cannot create moves when invoice state'
+                                    ' is \'%s\'.') % (state,))
+
+    @api.multi
+    def _check_existing_move(self):
+        """ Check existing move """
+        if self.move_id:
+            raise ValidationError(_('Move was already created.'))
+
+    @api.multi
+    def _check_init_entry(self):
```

Comment:
After backporting the spread logic from V11, I think that the `init_entry` field is not necessary anymore and could be removed, as well as this `_check_init_entry()` method. Could you check?

Same discourse of the field `type = fields.Selection()`.